File: Jung-jaeho/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <utility>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    int cnt = 0;
    int delay = 0;
    int flag = 0;
    //작업진도 100 , 작업속도 100

    while(1){
        cnt=0;
        delay++;
        if(flag == progresses.size())
            break;

        for (int i = flag; i < progresses.size(); i++){
            if ((progresses[flag] + (speeds[flag] * delay)) >= 100){
                flag++;
                cnt++;
            }
        }
        if (cnt != 0)
            answer.push_back(cnt);
    }
    return answer;
}
int main(){
    vector<int> progresses={93,30,55};
    vector<int> speeds = {1,30,5};
    vector<int> answer;

    answer=solution(progresses,speeds);


}

File: 취업한사람들/zojae031/Algorithm/3Week/src/기능개발.kt
//import java.util.*
//
//fun main() {
//    Solution().solution(intArrayOf(93, 30, 55), intArrayOf(1, 30, 5))
//}
//
//class Solution {
//    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
//        var answer = mutableListOf<Int>()
//
//        val list: Queue<Int> = LinkedList<Int>()
//        //List 초기화
//        progresses.map {
//            //7 70 45 로 만들어주는 작업
//            SCORE - it
//        }.also {
//            // 7 2 9 로 만들어주는 작업
//            for (i in progresses.indices) {
//                list.add(it[i] / speeds[i])
//            }
//        }
//
//        while (list.isNotEmpty()) {
//            val head = list.remove()
//            var cnt = 1
//            while (list.isNotEmpty() && head >= list.peek()) {
//                cnt++
//                list.remove()
//            }
//            answer.add(cnt)
//        }
//        return answer.toIntArray()
//    }
//
//    companion object {
//        const val SCORE = 100
//    }
//}